### ISSUE TITLE
Ignore-comments don't work for multi-line results

### ISSUE THREAD

**Issue by natlibfi-arlehiko** [ajinabraham/njsscan#24]
Ignoring rules using comments doesn't seem to work for multi-line findings:
# Works
## Code
```js
await connection.query(`INSERT INTO records VALUES ${generatePlaceholders(values.length)}`, values); //ignore: node_sqli_injection
```

# Doesn't work
## Code
```js
const values = chunk.map(({
  identifier,
  record
}) => [identifier, record]).flat();

await connection.query(`INSERT INTO records VALUES ${generatePlaceholders(values.length)}`, values); //ignore: node_sqli_injection
```
## Error
```
======================================================================================================
RULE ID: node_sqli_injection
OWASP: A1: Injection
CWE: CWE-89: Improper Neutralization of Special Elements used in an SQL Command ('SQL Injection')
DESCRIPTION: Untrusted input concatinated with raw SQL query can result in SQL Injection.
SEVERITY: ERROR
======================================================================================================

__________________FILES___________________________


File: /test.js
Match Position: 7 - 100
Line Number(s): 6
Match String: await connection.query(`INSERT INTO records VALUES ${generatePlaceholders(values.length)}`, values); //ignore: node_sqli_injection
```

**Comment by ajinabraham** [comment]
Thanks for reporting. I will take a look.

**Comment by ajinabraham** [comment]
This is due to an external bug in semgrep that matches the same line twice and `//ignore: node_sqli_injection` removes only one of the findings.

If you remove the comment, you can see that there is two findings along same lines

```
RULE ID: node_sqli_injection
OWASP: A1: Injection
CWE: CWE-89: Improper Neutralization of Special Elements used in an SQL Command ('SQL Injection')
DESCRIPTION: Untrusted input concatinated with raw SQL query can result in SQL Injection.
SEVERITY: ERROR
======================================================================================================

__________________FILES___________________________


File: test.js
Match Position: 1 - 101
Line Number(s): 1: 6
Match String: const values = chunk.map(({
  identifier,
  record
}) => [identifier, record]).flat();

await connection.query(`INSERT INTO records VALUES ${generatePlaceholders(values.length)}`, values);


File: test.js
Match Position: 7 - 100
Line Number(s): 6
Match String: await connection.query(`INSERT INTO records VALUES ${generatePlaceholders(values.length)}`, values);
```

Tracked upstream https://github.com/ajinabraham/njsscan/issues/24

**Comment by ajinabraham** [comment]
There is also a bug in njsscan that is fixed in latest master. With the latest changes, you could ignore two matches with two comments like.

```
const values = chunk.map(({  //ignore: node_sqli_injection
  identifier,
  record
}) => [identifier, record]).flat();

await connection.query(`INSERT INTO records VALUES ${generatePlaceholders(values.length)}`, values); //ignore: node_sqli_injection
```

Please note these changes are not available in pypi yet.

**Comment by ajinabraham** [comment]
As per upstream comments. This is the expected behavior. There was two findings triggered by the rule. The fixes made to njsscan master allow you to suppress both findings as mentioned in the comments above.

**Comment by ajinabraham** [comment]
Fix is now published to pypi available from njsscan 0.1.2
